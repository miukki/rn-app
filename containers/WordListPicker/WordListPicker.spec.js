import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store'
import Immutable from 'immutable'
import WordSelectionPicker from './'

import wordsLists from '../../data/words_lists'

Enzyme.configure({ adapter: new Adapter() })
const shallow = Enzyme.shallow
jest.mock('../../redux-modules/wordlist/actions')

const mockStore = configureStore([])

describe('WordSelectionPicker container', function() {
  let component

  beforeAll(() => {
    const testState = Immutable.fromJS({
      studyPlan: {
        wordsLists,
        learnedWords: []
      }
    })
    const store = mockStore(testState)
    component = shallow(<WordSelectionPicker />, { context: { store } })
  })

  it('should map state to props', () => {
    expect(component.props().items).toMatchSnapshot()
  })
})
