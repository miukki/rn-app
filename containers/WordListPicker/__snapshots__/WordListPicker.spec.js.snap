// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordSelectionPicker container should map state to props 1`] = `
Array [
  Object {
    "onPress": [Function],
    "progress": 0,
    "title": "IELTS Listening",
  },
  Object {
    "onPress": [Function],
    "progress": 0,
    "title": "IELTS Speeking",
  },
]
`;
